Feature/post details

aleesteele opened the pull request:
closes #26 #29 #27 (new branch, replacing #94 )

Summary:
The `post__details` component is a `<ul>` element, which was put directly in `post.html`. Each include, `post-details.html`, `post-downloads.html` and `post-related.html` is enclosed in an li element. The markup should follow the format that @jnschrag and @tuckabelle prefer, which is posted below for reference. 

A few notes on stylistic choices:
- The if loop, `{% if detail[1].size > 0 %}` in `post-details.html` was made to account for any labels that may not have a value. For example, it prevents 'Downloads' from populating the 'Details' section on the 'Workshop for Workshopping' page because it has no value. However, this might be a bug, but the `size` filter doesn't count dates as characters, so with this filter along, it doesn't print the `event date` value on workshop pages. For this reason, a separate if loop was made to detect if `page.collection == 'workshops'`, which prints the date detail separately.
- On tablet, the details component is styled using flexbox. Should this feature be dependent on the post collection? I can see how the flex styles are somewhat unnecessary on shorter pages, like Workshops. 
- Certain styles had to be added manually (like for `<li>` in `post__details-item`), as using the placeholder for `%labeltext__list` seemed to be overridden by the parent class. At least, this seems to be the case on my computer - I know that I've stylistically had certain things happen that may not appear on other computers.
- One class was made for the downloads/related-posts hover styles, because they have the same features.

I hope these text descriptions are clear. If it's easier to go over anything in person, please let me know!

For easier reference, this is the semantic markup that @jnschrag suggested in the previous PR:
```<ul class="post__details-list">
  <li class="post__details-item">
    Categories
    <ul class="post-details">
      <li>Categories</li>
      <li>Categories</li>
    </ul>
  </li>
  <li class="post__details-item">
    Details
    <ul class="post-details">
      <li>
        Type
        <ul>
          <li>Platform Control and Informational Support</li>
          <li>Secondary Type</li>
        </ul>
      </li>
      <li>
        Characteristics
        <ul>
          <li>Detectibility</li>
          <li>Speed</li>
        </ul>
      </li>
    </ul>
  </li>
  <li class="post__details-item">
    Download
    <ul class="post-details">
      <li>Download 1</li>
      <li>Download 2</li>
    </ul>
  </li>
</ul>```

jnschrag commented on _includes/post-downloads.html:
```diff
@@ -0,0 +1,15 @@
+  <li class="post__details-item">
+    <ul class="post-details">
+      {% for download in downloads %}
+        <li class="post__details-styled"><a href={{ download.media_upload }}>{{ download.label }} <i class="icon icon-download"></i></a></li>
```
The URL should be wrapped in quotation marks.

aleesteele commented:
- The <ul> was originally included in the main `post.html`, enclosing all three includes. This was done with the thought that the <li> elements of each include would be enclosed in a single `post__details-list` class. Open further reflection, because this is a ul class and not an id, this approach didn't make sense. Also, it doesn't really seem to matter whether or not they're all enclosed in a single <ul> element. **This logic has been revised. Each component is enclosed in the `post__details-list` parent <ul> element so that the titles are enclosed in the proper class. **
- Fixed the margin/padding errors. Added `:not(:first-child)` and `:not(:last-child)` to account for edge cases, when padding at the top/bottom of the div is unnecessary.

aleesteele commented:
@jnschrag @leezaweb Added a bit of margin - should be good to go otherwise!
